libs/sdk-core/src/persist/transactions.rs

```diff
@@ -233,11 +238,16 @@ impl SqliteStorage {
     fn sql_row_to_payment(&self, row: &Row) -> PersistResult<Payment, rusqlite::Error> {
         let payment_type_str: String = row.get(1)?;
         let amount_msat = row.get(3)?;
+        let status: PaymentStatus = row.get(5)?;
+        let failed_amount_msat: Option<u64> = row.get(12)?;
         let mut payment = Payment {
             id: row.get(0)?,
             payment_type: PaymentType::from_str(payment_type_str.as_str()).unwrap(),
             payment_time: row.get(2)?,
-            amount_msat,
+            amount_msat: match status {
+                PaymentStatus::Failed => failed_amount_msat.unwrap_or_default(),
```

Comment:
Don't we have a migration issue for existing failed payments (even with non zero invoices) that currently will get zero value here?

Reply:
True, it would be 0 for previously failed payments. To my knowledge its not possible to retreive failed payment amounts from GL. The alternative is to add failed_amount_msat: Option<u64> to Payment directly. But it will still not have a value for a failed payment.

Reply:
I am not concerned about existing failed payments of zero invoices. We can't control these but I am worried about existing failed payments of invoices with amount. Shouldn't we fall back to the existing invoice amount to solve this?

Reply:
Ah, that's a good idea. But it is possible as a migration step? I'll look into it